scipy/stats/tests/test_discrete_distns.py

```diff
@@ -112,3 +117,142 @@ def test_skellam_gh11474():
+
+        @np.vectorize
+        def pmf_mean_var(x, N, m1, n, w):
+            # simple implementation of FNCH pmf
+            m2 = N - m1
+            xl = np.maximum(0, n-m2)
+            xu = np.minimum(n, m1)
+
+            def f(x):
+                t1 = binom(m1, x)
+                t2 = binom(m2, n - x)
+                return t1 * t2 * w**x
+
+            def P(k):
+                return sum((f(y)*y**k for y in range(xl, xu + 1)))
+
+            P0 = P(0)
+            P1 = P(1)
+            P2 = P(2)
+            pmf = f(x) / P0
+            mean = P1 / P0
+            var = P2 / P0 - (P1 / P0)**2
+            return pmf, mean, var
+
+        pmf, mean, var = pmf_mean_var(x, N, m1, n, odds)
+        assert_allclose(fnch.pmf(x, N, m1, n, odds), pmf)
+        assert_allclose(fnch.stats(N, m1, n, odds, moments='m'), mean)
+        assert_allclose(fnch.stats(N, m1, n, odds, moments='v'), var)
+
+    def test_wnch_naive(self):
+        # test against a very simple implementation
+
+        np.random.seed(2)
+        shape = (2, 4, 3)
+        max_m = 100
+        m1 = np.random.randint(1, max_m, size=shape)
+        m2 = np.random.randint(1, max_m, size=shape)
+        N = m1 + m2
+        n = randint.rvs(0, N, size=N.shape)
+        xl = np.maximum(0, n-m2)
+        xu = np.minimum(n, m1)
+        x = randint.rvs(xl, xu, size=xl.shape)
+        w = np.random.rand(*x.shape)*2
+
+        def support(N, m1, n, w):
+            m2 = N - m1
+            xl = np.maximum(0, n-m2)
+            xu = np.minimum(n, m1)
+            return xl, xu
+
+        @np.vectorize
+        def mean(N, m1, n, w):
+            m2 = N - m1
+            xl, xu = support(N, m1, n, w)
+
+            def fun(u):
+                return u/m1 + (1 - (n-u)/m2)**w - 1
+
+            return root_scalar(fun, bracket=(xl, xu)).root
+
+        assert_allclose(wnch.mean(N, m1, n, w), mean(N, m1, n, w), rtol=2e-2)
+
+        @np.vectorize
+        def variance(N, m1, n, w):
+            m2 = N - m1
+            u = mean(N, m1, n, w)
+            a = u * (m1 - u)
+            b = (n-u)*(u + m2 - n)
+            return N*a*b / ((N-1) * (m1*b + m2*a))
+
+        assert_allclose(wnch.stats(N, m1, n, w, moments='v'),
+                        variance(N, m1, n, w), rtol=5e-2)
+
+        @np.vectorize
+        def pmf(x, N, m1, n, w):
+            m2 = N - m1
+            xl, xu = support(N, m1, n, w)
+
+            def integrand(t):
+                D = w*(m1 - x) + (m2 - (n-x))
+                res = (1-t**(w/D))**x * (1-t**(1/D))**(n-x)
+                return res
+
+            def f(x):
+                t1 = binom(m1, x)
+                t2 = binom(m2, n - x)
+                the_integral = quad(integrand, 0, 1,
+                                    epsrel=1e-16, epsabs=1e-16)
+                return t1 * t2 * the_integral[0]
+
+            return f(x)
+
+        pmf0 = pmf(x, N, m1, n, w)
+        pmf1 = wnch.pmf(x, N, m1, n, w)
```

Comment:
The naive PMF calculation is very unreliable. It matches very well in some cases and fails miserably in others. It's clear that `wnch` does the right thing and the `pmf` function above is wrong. I don't think it's appropriate to hand-tuning the test to find only examples for which the two agree, so the strategy here is to perform the calculations for all the randomly-generated inputs used for `fnch` above, but just discount the cases where `pmf` is clearly unreliable (the PMFs do not sum to 1).